core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java

```diff
@@ -165,10 +169,19 @@ void reconnect(String transactionServiceGroup) {
             return;
         }
         if (CollectionUtils.isEmpty(availList)) {
-            String serviceGroup = RegistryFactory.getInstance()
-                                                 .getServiceGroup(transactionServiceGroup);
-            LOGGER.error("no available service '{}' found, please make sure registry config correct", serviceGroup);
-            return;
+            RegistryService registryService = RegistryFactory.getInstance();
+            String clusterName = registryService.getServiceGroup(transactionServiceGroup);
+
+            if (StringUtils.isBlank(clusterName)) {
+                LOGGER.error("can not get cluster name in registry config '{}{}', please make sure registry config correct",
+                        ConfigurationKeys.SERVICE_GROUP_MAPPING_PREFIX,
+                        transactionServiceGroup);
+                return;
+            }
+
+            if (!(registryService instanceof FileRegistryServiceImpl)) {
+                LOGGER.error("no available service found, please make sure registry config correct and keep your seata server running");
+            }
```

Comment:
`return;`

Reply:
> line 185: `return;`
> If missed, it may throw NPE.

it's necessary? the method _getAvailServerList_  will return a empty list

Reply:
The current method should not consider how the 'getAvailServerList' is implemented.
If the codes of 'getAvailServerList' changed, it will throw NPE.

Reply:
In my opinion, it is better for the `getAvailServerList` method to return null if the list is empty.

Reply:
> The current method should not consider how the `getAvailServerList` is implemented.
> If the codes of `getAvailServerList` changed, it will throw NPE.

fixed,please review it